Fix symlinks in bazel-release

jsafrane opened the pull request:
kubernetes.tar.gz produced by `make bazel-release` should contain all symlinks that are in `cluster/` subdirectory. These symlinks are present in kubernetes.tar.gz created by `make quick-release`.

/kind bug
Fixes: #78940

```release-note
NONE
```

jsafrane commented:
/test pull-kubernetes-e2e-gce-iscsi

jsafrane commented:
/test pull-kubernetes-e2e-gce-iscsi

jsafrane commented:
/test pull-kubernetes-e2e-gce-iscsi

jsafrane commented:
/test pull-kubernetes-e2e-gce-iscsi

jsafrane commented:
/test pull-kubernetes-e2e-gce-iscsi-serial

jsafrane commented:
/test pull-kubernetes-e2e-gce-iscsi

jsafrane commented:
/test pull-kubernetes-e2e-gce-iscsi-serial

jsafrane commented:
CC @ixdy, most of `release-tars/BUILD` is your work.

jsafrane commented:
If this works, `pull-kubernetes-e2e-gce-iscsi` should succeed.

cblecker commented:
/assign

cblecker commented:
/test pull-kubernetes-integration

cblecker commented:
/hold

So this works.. sort of. The symlinks are created, but there's a permissions problem with the symlinks and cross building. I'm investigating.

BenTheElder commented:
/uncc @ixdy
/cc @fejta

justaugustus commented:
/area release-eng
/milestone v1.16
/priority important-soon

cblecker commented:
Update: still poking at this. Unfortunately the symlinks are broken in certain circumstances.. trying to isolate it.

fejta reviewed (approved):
/lgtm
/approve

cblecker commented:
Found the issue. This needs to be brought in as well: https://github.com/kubernetes/repo-infra/pull/124

cblecker commented:
@jsafrane Can you add in the below patch to bring in the fix (https://github.com/kubernetes/repo-infra/compare/f85734f673056977d8ba04b0386394b684ca2acb...9f4571ad7242bf3ec4b47365062498c2528f9a5f)?:

```patch
diff --git a/build/root/WORKSPACE b/build/root/WORKSPACE
index cb05ed3f77ccc..53588303faa89 100644
--- a/build/root/WORKSPACE
+++ b/build/root/WORKSPACE
@@ -37,9 +37,9 @@ versions.check(minimum_bazel_version = "0.23.0")
 
 http_archive(
     name = "io_k8s_repo_infra",
-    sha256 = "462dcc769f29e9eee16499edb818e2fc2ffc24efea28300d53bbc098d27d322c",
-    strip_prefix = "repo-infra-f85734f673056977d8ba04b0386394b684ca2acb",
-    urls = mirror("https://github.com/kubernetes/repo-infra/archive/f85734f673056977d8ba04b0386394b684ca2acb.tar.gz"),
+    sha256 = "f6d65480241ec0fd7a0d01f432938b97d7395aeb8eefbe859bb877c9b4eafa56",
+    strip_prefix = "repo-infra-9f4571ad7242bf3ec4b47365062498c2528f9a5f",
+    urls = mirror("https://github.com/kubernetes/repo-infra/archive/9f4571ad7242bf3ec4b47365062498c2528f9a5f.tar.gz"),
 )
 
 http_archive(
```

jsafrane committed:
Fix symlinks in bazel-release

kubernetes.tar.gz should contain all symlinks that are in cluster/
subdirectory.

jsafrane committed:
Update WORKSPACE to fix symlink permissions

Updating to get https://github.com/kubernetes/repo-infra/pull/124

jsafrane commented:
@cblecker, added as separate commit

cblecker commented:
/lgtm
/approve
/hold cancel
/retest

cblecker commented:
/test pull-kubernetes-e2e-gce-iscsi
/test pull-kubernetes-e2e-gce-iscsi-serial
